tests/cpp/test_segmentation.cpp

```diff
@@ -578,4 +578,33 @@ TEST_F(SegmentationTest, SliceSegmentCasts) {
   testValidate(fec.fusion(), outputs, {in0}, __LINE__, __FILE__);
 }
 
+TEST_F(SegmentationTest, codeGenSupportedMergeIssue1970) {
+  auto fusion = std::make_unique<Fusion>();
+  FusionGuard fg(fusion.get());
+
+  auto tv0 = makeSymbolicTensor(3, DataType::Half);
+  fusion->addInput(tv0);
+  auto tv_bias = makeSymbolicTensor(1, DataType::Half);
+  fusion->addInput(tv_bias);
+
+  auto* tv1 = castOp(DataType::Float, tv0);
+  auto* tv2 = sum(tv1, {0, 2});
+  auto* tv3 = castOp(DataType::Float, tv_bias);
+  auto* tv4 = add(tv2, tv3);
+  auto* tv5 = castOp(DataType::Half, tv4);
+  fusion->addOutput(tv5);
+  auto* tv6 = add(tv3, tv3);
```

Comment:
The trick here is that the segment has two uses of forwarded inputs.

Reply:
Thanks! Can the test be simplified? Maybe a chain of unary followed by `add(x,x)` with a segment_set at the end? The segment_set is to skip the initial attempt to schedule the whole fusion unsegmented, a trick used by other tests in this file.